packages/frontend/src/components/SwapPath/SwapPath.tsx

```diff
@@ -0,0 +1,111 @@
+import { motion, AnimatePresence } from 'framer-motion';
+import { EitherQuoteSifiAction, Quote } from '@sifi/sdk';
+import { ReactComponent as UniswapIcon } from 'src/assets/bridges/uniswap.svg';
+import { ReactComponent as CurveIcon } from 'src/assets/bridges/curve.svg';
+import { ReactComponent as StargateIcon } from 'src/assets/bridges/stargate.svg';
+import { ReactComponent as ArrowRight } from 'src/assets/arrow-right.svg';
+
+type SwapPathProps = {
+  quote: Quote | null;
+};
+
+type StepDetails = {
+  [key: string]: { icon: JSX.Element; name: string };
+};
+
+type Step = {
+  icon: JSX.Element;
+  name: string;
+};
+
+const iconClassName = 'w-4 h-4 border-smoke border rounded-full';
+
+const stepDetails: StepDetails = {
+  uniswap: { icon: <UniswapIcon className={iconClassName} />, name: 'Uniswap' },
+  stargate: { icon: <StargateIcon className={iconClassName} />, name: 'Stargate' },
+  curve: { icon: <CurveIcon className={iconClassName} />, name: 'Curve' },
+};
+
+type StepInfo = {
+  icon: JSX.Element;
+  name: string;
+};
+
+const getStepDetailsFromAction = (action: EitherQuoteSifiAction): StepInfo | null => {
+  if ('exchange' in action) {
+    const exchange = action.exchange.toLowerCase();
+
+    for (let key in stepDetails) {
+      const regex = new RegExp(key, 'i');
+
+      if (regex.test(exchange)) {
+        const detail = stepDetails[key];
+
+        if (detail) {
+          return {
+            icon: detail.icon,
+            name: detail.name,
+          };
+        }
+      }
+    }
+  }
+  return null;
+};
+
+const SwapPath: React.FC<SwapPathProps> = ({ quote }) => {
+  if (!quote || !('element' in quote.source.quote) || !quote.source.quote.element.actions) {
+    return null;
+  }
+
+  let steps: Step[] = [];
+
+  quote.source.quote.element.actions.forEach(action => {
```

Comment:
I personally feel like it would be cleaner to use `Array.reduce()` here like `const steps = quote.source.quote.element.actions.reduce()` instead of looping over the array and pushing them onto the final array

Reply:
Well, this PR just separates the component, this logic was written prior.